src/quo/components/token.cljs

```diff
@@ -0,0 +1,53 @@
+(ns quo.components.token
+  (:require [clojure.string :as string]
+            [quo.foundations.resources :as resources]
+            [react-native.core :as rn]
+            [utils.number]))
+
+(defn- size->number
+  "Remove `size-` prefix in size keywords and returns a number useful for styling."
+  [k]
+  (utils.number/parse-int (subs (name k) 5)))
+
+(defn- token-style
+  [style size]
+  (let [size-number (if (keyword? size)
+                      (size->number size)
+                      size)]
+    (assoc style
+           :width  size-number
+           :height size-number)))
+
+(defn- get-token-image*
+  [token]
+  (let [token-name (condp #(%1 %2) token
+                     string?  (keyword (string/lower-case token))
+                     keyword? (keyword (string/lower-case (name token)))
+                     nil)]
+    (resources/get-token token-name)))
+
+(def ^:private get-token-image (memoize get-token-image*))
+(def ^:private b64-png-image-prefix "data:image/png;base64,")
+
+(defn view
+  "Render a token image.
+   Props:
+   - style:           extra styles to apply to the `rn/image` component.
+   - size:            `:size-nn` or just `nn`, being `nn` any number. Defaults to 32.
+   - token:           string or keyword, it can contain upper case letters or not.
+                      E.g. all of these are valid and resolve to the same:
+                      :token/snt | :snt | :SNT | \"SNT\" | \"snt\".
+   - image-source:    Ignores `token` and uses this as parameter to `rn/image`'s source.
+   - add-b64-prefix?: If true, adds `data:image/png;base64,` as prefix to the string
+                      passed as `image-source`.
+  "
+  [{:keys [token size style image-source add-b64-prefix?]
+    :or   {size  32
+           token :snt}}]
```

Comment:
don't think it's a good idea to fallback to snt when token is not provided/defined
this may cause user confusion or even asset loss

Reply:
I agree, will remove it.

BTW, designers are pending to create a default component when no image is available